flexmeasures/api/v3_0/sensors.py

```diff
@@ -414,8 +414,8 @@ def trigger_schedule(  # noqa: C901
             soc_min=soc_min,
             soc_max=soc_max,
             roundtrip_efficiency=roundtrip_efficiency,
-            up_deviation_price_sensor=up_deviation_price_sensor,
```

Comment:
Yes. Any sensor expressing some (to be minimized) intensity with respect to energy is a possible use case.

Note that, currently, there is no check on units of these "price" sensors. I could imagine it makes sense to check the unit's denominator, and raise an error if it is not the same unit as that of the to be scheduled (power) sensor, or a first degree time integral thereof (e.g. kWh).

Reply:
Looks lie a follow-up issue!

Q: Why should it be the same unit as the (power) sensor? It needs some quantity assigned to the scheduled quantity...

Anyway, to get a feeling for what we're after, here are two examples I'd deem valid (scheduled sensor -> pricing sensor)

kWh -> EUR/kWh
kWh -> CO2/MW

What would be invalid?

kWh -> EUR/W
kW -> EUR/kWh  (is kW even valid to be schduled?)
kWh -> l/s

Of course, via the other parameter we also price the inflexible sensors, so they should also be checked.

Reply:
A: I wrote that the **denominator** of the intensity unit should be the same as the unit of the schedulable sensor (or time convertible), but I have further thoughts about that below.

Our scheduled sensors actually typically record flows, so our most typical case is kW -> EUR/kWh. (We first compute the change in kWh stock by multiplying the power sensor values by the resolution.) I believe it would be valid to swap out EUR for anything else worth minimizing, and I'd also want to support any combination of metric prefixes (W, kW, MW).

In general, I'd start with enforcing the assumption that the **denominator** of the price sensor is \<unit of time\> times the unit of the power sensor. And I'd allow anything as the **numerator** of the price sensor.

---

I now consider the below to be out of scope (i.e. "invalid" for now), but maybe you find it informative:

### kWh -> EUR/kWh

To understand this case, I think of EUR/kWh as the ratio between a EUR/h money flow sensor with respect to a kW power flow sensor.

If we consider kWh sensors to be valid, we should make clear that we assume that they represent stock changes (total prosumption over some time interval) instead of a cumulative stock (e.g. a *meterstand* or SoC). Alternatively, we let the user inform us to first compute the change in kWh stock by computing deltas for the energy sensor.

### kW -> EUR/kW

The price suggests that the stock is something measured in kW rather than kWh (e.g. activated reserve capacity). We should make clear that we assume that the power sensor represents stock changes (activated reserve capacity, such as a reduction in load) instead of a cumulative stock (e.g. actual load).

### kW/h -> EUR/kW

To understand this case, I think of EUR/kW as the ratio between a EUR/h money flow sensor with respect to a kW/h ramp rate sensor.

Here, the schedulable sensor is recording ramp rates. We first compute the change in kW stock by multiplying the ramp rate sensor's values by its resolution.

Reply:
Discussion continues in #483.